tests/components/group/test_notify.py

```diff
@@ -1,98 +1,83 @@
+        await async_setup_component(
+            hass,
+            notify.DOMAIN,
+            {
+                "notify": [
+                    {"name": "demo1", "platform": "demo"},
+                    {"name": "demo2", "platform": "demo"},
+                ]
+            },
+        )
+        await hass.async_block_till_done()
 
-        self.service = asyncio.run_coroutine_threadsafe(
-            group.async_get_service(
-                self.hass,
+    service = await group.async_get_service(
+        hass,
+        {
+            "services": [
+                {"service": "demo1"},
                 {
-                    "services": [
-                        {"service": "demo1"},
-                        {
-                            "service": "demo2",
-                            "data": {
-                                "target": "unnamed device",
-                                "data": {"test": "message"},
-                            },
-                        },
-                    ]
+                    "service": "demo2",
+                    "data": {
+                        "target": "unnamed device",
+                        "data": {"test": "message"},
+                    },
                 },
-            ),
-            self.hass.loop,
-        ).result()
-
-        assert self.service is not None
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    def test_send_message_with_data(self):
-        """Test sending a message with to a notify group."""
-        asyncio.run_coroutine_threadsafe(
-            self.service.async_send_message(
-                "Hello", title="Test notification", data={"hello": "world"}
-            ),
-            self.hass.loop,
-        ).result()
-        self.hass.block_till_done()
-
-        assert self.service1.send_message.mock_calls[0][1][0] == "Hello"
-        assert self.service1.send_message.mock_calls[0][2] == {
-            "title": "Test notification",
-            "data": {"hello": "world"},
-        }
-        assert self.service2.send_message.mock_calls[0][1][0] == "Hello"
-        assert self.service2.send_message.mock_calls[0][2] == {
-            "target": ["unnamed device"],
-            "title": "Test notification",
-            "data": {"hello": "world", "test": "message"},
-        }
+            ]
+        },
+    )
+
+    """Test sending a message with to a notify group."""
+    await service.async_send_message(
+        "Hello", title="Test notification", data={"hello": "world"}
+    )
+
+    await hass.async_block_till_done()
+
+    assert service1.send_message.mock_calls[0][1][0] == "Hello"
+    assert service1.send_message.mock_calls[0][2] == {
+        "title": "Test notification",
+        "data": {"hello": "world"},
+    }
+    assert service2.send_message.mock_calls[0][1][0] == "Hello"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)